### PR TITLE
eww: add support for svg icons

### PR DESCRIPTION
**Pull request by fufexan**
## Description of changes

Allow themes like Papirus to be displayed correctly, either directly inside eww or inside `eww inspector`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/eww/default.nix [hunk 1]
```diff
@@ -22,9 +24,9 @@ rustPlatform.buildRustPackage rec {
 
   cargoHash = "sha256-dC7yVJdR7mO0n+sxWwigM1Q4tbDv5ZuOINHHlUIPdA0=";
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ pkg-config wrapGAppsHook ];
 
-  buildInputs = [ gtk3 gdk-pixbuf ] ++ lib.optional withWayland gtk-layer-shell;
+  buildInputs = [ gtk3 gdk-pixbuf librsvg ] ++ lib.optional withWayland gtk-layer-shell;
```
Comment (figsoda) [inline pkgs/applications/window-managers/eww/default.nix:29]: does that mean we can remove gdk-pixbuf here?

File: pkgs/applications/window-managers/eww/default.nix [hunk 2]
```diff
@@ -6,6 +6,8 @@
 , gdk-pixbuf
 , withWayland ? false
 , gtk-layer-shell
+, librsvg
+, wrapGAppsHook
```
Comment (figsoda) [inline pkgs/applications/window-managers/eww/default.nix:10]: please move `wrapGAppsHook` to after `pkg-config`, and `librsvg` to before `withWayland`

> Arguments should be listed in the order they are used, with the exception of `lib`, which always goes first. (Last line in `CONTRIBUTING.md`)

File: pkgs/applications/window-managers/eww/default.nix (hunk 1 above)
Comment (fufexan) [reply pkgs/applications/window-managers/eww/default.nix:29]: Probably not, as that handles PNG/JPG/non-vector images.
I'll test and report.

Comment (fufexan) [reply pkgs/applications/window-managers/eww/default.nix:29]: There doesn't seem to be any difference, should I remove it?

Comment (figsoda) [reply pkgs/applications/window-managers/eww/default.nix:29]: yes please, thanks for testing!

**Review by figsoda — APPROVED** [review]

**Pull request merged**